pkg/transformers/mutatefield.go

```diff
@@ -79,3 +81,10 @@ func mutateField(
 		return fmt.Errorf("%#v is not expected to be a primitive type", typedV)
 	}
 }
+
+func getFirstPathSegment(pathToField []string) (string, bool) {
+	if strings.HasSuffix(pathToField[0], "[]") {
```

Comment:
Not sure if we need to add support for escaping this. Will there ever be keys ending with `[]`? Probably not, but...